### PR TITLE
Clarify where importmap declaration must live and errors

### LINKED ISSUE #25430: Docs for <script type="importmap"> missing information about order
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap

### What specific section or headline is this issue about?

Exceptions

### What information was incorrect, unhelpful, or incomplete?

There is another type of error which can occur, when the importmap is loaded too late. 
On Chrome, I am getting this error:

```
An import map is added after module script load was triggered.
```

### What did you expect to see?

It is never mentioned in the docs where this should be placed. For example, should this be above any `type="module`"? What about `type="modulepreload"`? I expect to see limitations about when it is usable. In some cases, external tools might need modifying just to have the ability to add script tags above some auto-generated tags, which cause this feature not to work (at least on Chrome).

### Do you have any supporting links, references, or citations?

The [relevant HTML standard](https://html.spec.whatwg.org/multipage/webappapis.html#import-map) specifies:

> Import maps are currently disallowed once any module loading has started, or once a single import map is loaded. These restrictions might be lifted in future specification revisions.

As such I believe it should be noted in the MDN documentation as well.

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/script/type/importmap`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/script/type/importmap/index.md
* Last commit: https://github.com/mdn/content/commit/5c30b5e4e57d1a77ce56b88b3de1171d57562808
* Document last modified: 2023-03-09T08:24:08.000Z

</details>

### PR DESCRIPTION
**Pull request by hamishwillee**
The importmap docs weren't as clear as they should be about where the importmap should be declared.

@Josh-Cena Would appreciate your sanity check/review.

Fixes #25430

### CODE REVIEW

File: files/en-us/web/html/element/script/type/importmap/index.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ The **`importmap`** value of the [`type`](/en-US/docs/Web/HTML/Element/script/ty
 An import map is a JSON object that allows developers to control how the browser resolves module specifiers when importing [JavaScript modules](/en-US/docs/Web/JavaScript/Guide/Modules).
 It provides a mapping between the text used as the module specifier in an [`import` statement](/en-US/docs/Web/JavaScript/Reference/Statements/import) or [`import()` operator](/en-US/docs/Web/JavaScript/Reference/Operators/import), and the corresponding value that will replace the text when resolving the specifier.
 The JSON object must conform to the [Import map JSON representation format](#import_map_json_representation).
-
+  
+Import maps are used to resolve module specifiers in static and dynamic imports, and therefore must be processed before any `<script>` elements that import modules.
```
Comment (hamishwillee) [inline files/en-us/web/html/element/script/type/importmap/index.md:16]:
```suggestion
Import maps are used to resolve module specifiers in static and dynamic imports, and therefore must be delcared and processed before any `<script>` elements that import modules.
```

File: files/en-us/web/html/element/script/type/importmap/index.md [hunk 2]
```diff
@@ -12,7 +12,8 @@ The **`importmap`** value of the [`type`](/en-US/docs/Web/HTML/Element/script/ty
 An import map is a JSON object that allows developers to control how the browser resolves module specifiers when importing [JavaScript modules](/en-US/docs/Web/JavaScript/Guide/Modules).
 It provides a mapping between the text used as the module specifier in an [`import` statement](/en-US/docs/Web/JavaScript/Reference/Statements/import) or [`import()` operator](/en-US/docs/Web/JavaScript/Reference/Operators/import), and the corresponding value that will replace the text when resolving the specifier.
 The JSON object must conform to the [Import map JSON representation format](#import_map_json_representation).
-
+  
+An import map is used to resolve module specifiers in static and dynamic imports, and therefore must be declared and processed before any `<script>` elements that import modules using specifiers declared in this map.
```
Comment (hamishwillee) [inline files/en-us/web/html/element/script/type/importmap/index.md:16]:
```suggestion
An import map is used to resolve module specifiers in static and dynamic imports, and therefore must be declared and processed before any `<script>` elements that import modules using specifiers declared in the map.
```

**Comment by hamishwillee** [comment]
Thanks @Josh-Cena - Accepted your suggestion. Good to merge?

**Review by Josh-Cena — APPROVED** [review]
Yes, LGTM modulo the lint errors

File: files/en-us/web/html/element/script/type/importmap/index.md [hunk 3]
```diff
@@ -12,7 +12,8 @@ The **`importmap`** value of the [`type`](/en-US/docs/Web/HTML/Element/script/ty
 An import map is a JSON object that allows developers to control how the browser resolves module specifiers when importing [JavaScript modules](/en-US/docs/Web/JavaScript/Guide/Modules).
 It provides a mapping between the text used as the module specifier in an [`import` statement](/en-US/docs/Web/JavaScript/Reference/Statements/import) or [`import()` operator](/en-US/docs/Web/JavaScript/Reference/Operators/import), and the corresponding value that will replace the text when resolving the specifier.
 The JSON object must conform to the [Import map JSON representation format](#import_map_json_representation).
-
+
```
Comment (hamishwillee) [inline files/en-us/web/html/element/script/type/importmap/index.md:15]:
```suggestion

```

File: files/en-us/web/html/element/script/type/importmap/index.md [hunk 4]
```diff
@@ -28,14 +29,16 @@ For more information, see the [Importing modules using import maps](/en-US/docs/
...
 Only the first import map in the document with an inline definition is processed; any additional import maps and external import maps are ignored.
-An [`error` event](/en-US/docs/Web/API/Element/error_event) is fired at script elements with `type="importmap"` that are not processed (are ignored).
 
 ### Exceptions
 
 - `TypeError`
   - : The import map definition is not a JSON object, the `importmap` key is defined but its value is not a JSON object, or the `scopes` key is defined but its value is not a JSON object.
 
 Browsers generate console warnings for other cases where the import map JSON does not conform to the [import map](#import_map_json_representation) schema.
+
```
Comment (hamishwillee) [inline files/en-us/web/html/element/script/type/importmap/index.md:39]:
```suggestion

```

**Pull request merged**